test/files/neg/t7721.check

```diff
@@ -22,6 +22,24 @@ t7721.scala:48: warning: abstract type pattern B.this.Foo is unchecked since it
 t7721.scala:48: warning: abstract type pattern B.this.Bar is unchecked since it is eliminated by erasure
     case x: Foo with Bar with Concrete => x.bippy + x.barry + x.dingo + x.conco + x.bongo
             ^
+t7721.scala:12: warning: The outer reference in this type test cannot be checked at run time.
+    case x: Foo with Concrete => x.bippy + x.conco
+         ^
+t7721.scala:16: warning: The outer reference in this type test cannot be checked at run time.
+    case x: Concrete with Foo => x.bippy + x.conco
+         ^
+t7721.scala:20: warning: The outer reference in this type test cannot be checked at run time.
+    case x: Foo with Bar => x.bippy + x.barry
+         ^
+t7721.scala:40: warning: The outer reference in this type test cannot be checked at run time.
+    case x: Foo with Concrete => x.bippy + x.dingo + x.conco
+         ^
+t7721.scala:44: warning: The outer reference in this type test cannot be checked at run time.
+    case x: Concrete with Foo => x.bippy + x.dingo + x.conco
+         ^
+t7721.scala:48: warning: The outer reference in this type test cannot be checked at run time.
+    case x: Foo with Bar with Concrete => x.bippy + x.barry + x.dingo + x.conco + x.bongo
+         ^
```

Comment:
Are these unfortunate casualties or is this progress?

Reply:
Previously refinement types had no outer check (even if all the components had the same prefix). Here's why:

```
cala> class Outer { class A; class B; type T = A with B }
class Outer

scala> :power
Power mode enabled. :phase is at typer.
import scala.tools.nsc._, intp.global._, definitions._
Try :help or completions for vals._ and power._

scala> val o = new Outer
val o: Outer = Outer@51044708

scala> val t1 = typeOf[o.T].dealias
val t1: $r.intp.global.Type = o.A with o.B

scala> t1.prefix
val res1: $r.intp.global.Type = <notype>

scala> t1.asInstanceOf[RefinedType].parents.map(_.prefix)
val res8: List[$r.intp.global.Type] = List(o.type, o.type)
```

This patch recurses into refinement types and adds outer tests for each element.

So I'd classify this as progress, although some code may have come to rely on the the weaker existing behaviour.